homeassistant/components/ecobee/notify.py

```diff
@@ -13,19 +13,19 @@
 
 def get_service(hass, config, discovery_info=None):
     """Get the Ecobee notification service."""
-    data = hass.data[DOMAIN]
     index = config.get(CONF_INDEX)
-    return EcobeeNotificationService(data, index)
+    return EcobeeNotificationService(hass, index)
 
 
 class EcobeeNotificationService(BaseNotificationService):
     """Implement the notification service for the Ecobee thermostat."""
 
-    def __init__(self, data, thermostat_index):
+    def __init__(self, hass, thermostat_index):
         """Initialize the service."""
-        self.data = data
+        self.hass = hass
         self.thermostat_index = thermostat_index
 
     def send_message(self, message="", **kwargs):
         """Send a message."""
-        self.data.ecobee.send_message(self.thermostat_index, message)
+        data = self.hass.data[DOMAIN]
+        data.ecobee.send_message(self.thermostat_index, message)
```

Comment:
Thanks for the feedback, will look into that. Regarding YAML removal, do we have to go through the deprecation + scheduled removal process or can it be just removed since it's currently broken anyway?

Reply:
I think we can just make a breaking change without deprecation period.

Reply:
Pushed what I have so far for review since I'll be out of town for next 2 weeks and won't be able to work on this. I'll create a PR for the documentation change as well at some point.